File: Baekjoon/Level33/11723.cpp
#include<iostream>
#include<string> // for to_string, stoi
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    int S = 0;
    int M, num;
    string c;
    cin >> M;
    for (int i = 0; i < M; i++) {
        cin >> c;
        switch (c[1])
        {
        case 'd': // add
            cin >> num;
            S |= (1 << (num - 1));
            break;
        case 'e': // remove
            cin >> num;
            S &= ~(1 << (num - 1));
            break;
        case 'h': // check
            cin >> num;
            if ((S >> (num - 1)) & 1)
                cout << "1\n";
            else
                cout << "0\n";
            break;
        case 'o': // toggle
            cin >> num;
            S = ((S >> (num - 1)) & 1) ? S & ~(1 << (num - 1)) : S | (1 << (num - 1));
            break;
        case 'l': // all
            S = (1 << 21) - 1;
            break;
        case 'm': // empty
            S = 0;
            break;
        default:
            break;
        }

    }
    return 0;
}

File: Baekjoon/Level33/11723.py
import sys;input = sys.stdin.readline

M = int(input())
S = set()
for _ in range(M):
    command = input()
    if command[0] == 'a':
        if command[1] == 'd':
            S.add(int(command.split()[1]))
        else: # all
            S = set(range(1, 21))
    elif command[0] == 'r':
        num = int(command.split()[1])
        if num in S:
            S.remove(num)
    elif command[0] == 'c':
        print(1 if int(command.split()[1]) in S else 0)
    elif command[0] == 't':
        num = int(command.split()[1])
        if num in S:
            S.remove(num)
        else:
            S.add(num)
    else:# command[0] == 'e':
        S = set()



